Introduce the new user-facing Logs API as a placeholder

MSNev opened the pull request:
Part of https://github.com/open-telemetry/oteps/pull/265

lmolkova commented on specification/README.md:
```diff
@@ -29,6 +29,7 @@ path_base_for_github_subdir:
   - [Tracing](trace/api.md)
   - [Metrics](metrics/api.md)
   - [Logs](logs/README.md)
+    - [Public API](logs/public-api.md)
```
having a new API comes with a cost:

- cognitive overload due to multiple concepts representing a similar thing that ends up on the same exporter in the same format
- configuration complexity for 3 SDK pipelines

So it has clear drawbacks, but I don't see any problem it solves or benefits it brings.

PS: 2 of these concepts were enough to spend weeks bikeshedding, having 3 (3!!) concepts is not going to help us make progress.

MSNev replied:
Discussed in the Logs and Spec SIG and removed and folded into the existing bridge-api file.

tedsuo commented on specification/logs/public-api.md:
```diff
@@ -0,0 +1,40 @@
+# Public Logs API
```
fyi there's no need to use the word "public" in the spec, the term API is defined to be public.

trask commented on specification/logs/bridge-api.md:
````diff
@@ -44,6 +45,14 @@ graph TD
     B -->|Emit| C(LogRecord)
 ```
 
+## Public API Development
````
(similar to Enabled which is in status development)

```suggestion
## Instrumentation API

Status: Development
```

pellared commented on specification/logs/event-api.md:
```diff
@@ -30,6 +31,16 @@ The Event API consists of these main components:
 
+## Instrumentation API Development
+
+**Status**: [Development](../document-status.md)
```
nit: Not sure if this is needed. The whole document is "development"

pellared commented on specification/logs/event-sdk.md:
```diff
@@ -29,6 +30,14 @@ API that provides users with this functionally.
+
+Note!: We are currently in the process of defining a new [Instrumentation API](./bridge-api.md#instrumentation-api).
+
+The intent is that this new API will incorporate the current functionality of this existing API and once it is defined and implemented, the Event API usage will be migrated, deprecated, renamed and eventually removed.
+
+No further work is scheduled for the current Event SDK at this time.
```
```suggestion
The intent is that Logs SDK will incorporate the current functionality of this existing SDK and once it is defined and implemented, the Events SDK usage will be migrated, deprecated, renamed and eventually removed.

No further work is scheduled for the current Events SDK at this time.
```

MSNev replied:
Let me adjust this in all references.

MSNev replied:
I'm going to call it Instrumentation API for consistency though.

pellared commented on specification/logs/event-api.md:
```diff
@@ -30,6 +31,14 @@ The Event API consists of these main components:
+
+Note!: We are currently in the process of defining a new [Instrumentation API](./bridge-api.md#instrumentation-api).
+
+The intent is that this Instrumentation API will incorporate the current functionality of this existing API and once it is defined and implemented, the Event API usage will be migrated, deprecated, renamed and eventually removed.
```
nit

```suggestion
The intent is that this Instrumentation API will incorporate the current functionality of this existing API and once it is defined and implemented, the Events API usage will be migrated, deprecated, renamed and eventually removed.
```

pellared commented on specification/logs/event-api.md:
```diff
@@ -30,6 +31,14 @@ The Event API consists of these main components:
+
+The intent is that this Instrumentation API will incorporate the current functionality of this existing API and once it is defined and implemented, the Event API usage will be migrated, deprecated, renamed and eventually removed.
+
+No further work is scheduled for the current Event API definition at this time.
```
```suggestion
No further work is scheduled for the current Events API definition at this time.
```

pellared commented on specification/logs/bridge-api.md:
```diff
@@ -145,6 +146,16 @@ SHOULD be documented that instrumentation authors needs to call this API each
+This set of API functions will provide the capabalities needed to emit a
+`LogRecord` as is currently provided by
+
+* [Logs Bridge API](./bridge-api.md)
```
```suggestion
```

I think we agreed that for now we that the API just needs to be capable of emitting events (log records with `event.name` attribute).

pellared commented on specification/logs/bridge-api.md:
```diff
@@ -145,6 +146,15 @@ SHOULD be documented that instrumentation authors needs to call this API each
+**Status**: [Development](../document-status.md)
+
+This set of API functions will provide the capabalities needed to emit a
+`LogRecord` as is currently provided by
+
+* [Events API](./event-api.md)
```
```suggestion
`LogRecord` as is currently provided by [Events API](./event-api.md).
```

trask commented on specification/logs/bridge-api.md:
```diff
@@ -145,6 +146,13 @@ SHOULD be documented that instrumentation authors needs to call this API each
+
+**Status**: [Development](../document-status.md)
+
+This set of API functions will provide the capabalities needed to emit a
```
```suggestion
This set of API functions will provide the capabilities needed to emit a
```

trask commented on specification/logs/event-api.md:
```diff
@@ -30,6 +31,14 @@ The Event API consists of these main components:
 
+## Instrumentation API Development
+
+Note!: We are currently in the process of defining a new [Instrumentation API](./bridge-api.md#instrumentation-api).
```
```suggestion
Note!: We are currently in the process of defining a new [Logs Instrumentation API](./bridge-api.md#instrumentation-api).
```

trask commented on specification/logs/event-api.md:
```diff
@@ -30,6 +31,14 @@ The Event API consists of these main components:
+
+Note!: We are currently in the process of defining a new [Instrumentation API](./bridge-api.md#instrumentation-api).
+
+The intent is that this Instrumentation API will incorporate the current functionality of this existing API and once it is defined and implemented, the Events API usage will be migrated, deprecated, renamed and eventually removed.
```
```suggestion
The intent is that this Logs Instrumentation API will incorporate the current functionality of this existing Events API and once it is defined and implemented, the Events API usage will be migrated, deprecated, renamed and eventually removed.
```

trask commented on specification/logs/event-sdk.md:
```diff
@@ -29,6 +30,14 @@ API that provides users with this functionally.
 
+## Instrumentation API Development
+
+Note!: We are currently in the process of defining a new [Instrumentation API](./bridge-api.md#instrumentation-api).
```
```suggestion
Note!: We are currently in the process of defining a new [Logs Instrumentation API](./bridge-api.md#instrumentation-api).
```

trask commented on specification/logs/event-sdk.md:
```diff
@@ -29,6 +30,14 @@ API that provides users with this functionally.
+
+Note!: We are currently in the process of defining a new [Instrumentation API](./bridge-api.md#instrumentation-api).
+
+The intent is that Logs SDK will incorporate the current functionality of this existing SDK and once it is defined and implemented, the Events SDK usage will be migrated, deprecated, renamed and eventually removed.
```
```suggestion
The intent is that Logs SDK will incorporate the current functionality of this existing Events SDK and once it is defined and implemented, the Events SDK usage will be migrated, deprecated, renamed and eventually removed.
```

trask commented on specification/logs/sdk.md:
```diff
@@ -538,4 +539,8 @@ return a Failure result.
 
+## Instrumentation API
+
+Note!: We are currently in the process of defining a new [Instrumentation API](./bridge-api.md#instrumentation-api).
```
```suggestion
Note!: We are currently in the process of defining a new [Logs Instrumentation API](./bridge-api.md#instrumentation-api).
```

trask reviewed (approved):
(editorial suggestions only)

lmolkova commented on specification/logs/event-api.md:
```diff
@@ -30,6 +31,14 @@ The Event API consists of these main components:
 
+## Instrumentation API Development
+
+Note!: We are currently in the process of defining a new [Logs Instrumentation API](./bridge-api.md#instrumentation-api).
```
super-nit
```suggestion
> [!NOTE]
> We are currently in the process of defining a new [Logs Instrumentation API](./bridge-api.md#instrumentation-api).
```

will be rendered as 

> [!NOTE]
> We are currently in the process of defining a new [Logs Instrumentation API](./bridge-api.md#instrumentation-api).

lmolkova commented on specification/logs/event-sdk.md:
```diff
@@ -29,6 +30,14 @@ API that provides users with this functionally.
 
+## Instrumentation API Development
+
+Note!: We are currently in the process of defining a new [Logs Instrumentation API](./bridge-api.md#instrumentation-api).
```
```suggestion
> [!NOTE]
> We are currently in the process of defining a new [Logs Instrumentation API](./bridge-api.md#instrumentation-api).
```

lmolkova commented on specification/logs/sdk.md:
```diff
@@ -538,4 +539,8 @@ return a Failure result.
 
+## Instrumentation API
+
+Note!: We are currently in the process of defining a new [Logs Instrumentation API](./bridge-api.md#instrumentation-api).
```
```suggestion
> [!NOTE]
> We are currently in the process of defining a new [Logs Instrumentation API](./bridge-api.md#instrumentation-api).
```

tedsuo reviewed (approved):
Looks great, let's get this merged and add the API method.

MSNev commented:
@trask, @lmolkova, @pellared can you please check the latest commit, it (should) include
- all of your suggested changes (which I committed from github)
- As per discussion in the Maintainers meeting (general agreement of calling it the "Logs Instrumentation API"), updated the missed renames as per Trasks commits from last week.

Process taken: Merged changes from remote branch (which included the commits), updated missed renamings and pushed them back out (leading to loads of listed "commits" -- which should just be duplicative), I normally "cleanup" and squash the commits locally to keep it clean, but that requires a forced push to my branch and I didn't want to do that for this PR, hence the "messy" looking commits.

MSNev commented on specification/logs/event-sdk.md:
```diff
@@ -29,6 +30,15 @@ API that provides users with this functionally.
+> [!NOTE]
+> We are currently in the process of defining a new [Logs Instrumentation API](./bridge-api.md#logs-instrumentation-api).
+
+The intent is that Logs SDK will incorporate the current functionality of this existing Events SDK and once it is defined and implemented, the Events SDK usage will be migrated, deprecated, renamed and eventually removed.
```
This note is for anyone reading it between now and when we have the new Logs API functionality specified.

I am aware that there is active work occurring (as this has existed for several years) and there are some instrumentations that are using this today, I believe in JS, Java (Android) and potentially now in Python (for LLM).

I'm also trying to explicitly defer the exact approach that we take until we get to that point, so by using these enumerations it (hopefully) is also acting as a further warning to not take any dependency on it (if they have not already) and for anyone how has we will hopefully get feedback.
